daemon/task.go

```diff
@@ -103,44 +112,102 @@ func FormatTime(t time.Time) string {
 
 // Map the task onto a map[string]interface{}, using the given route for the Location()
 func (t *Task) Map(route *mux.Route) map[string]interface{} {
+	t.Lock()
+	defer t.Unlock()
+
 	return map[string]interface{}{
 		"resource":   t.Location(route),
 		"status":     t.State(),
-		"created_at": FormatTime(t.CreatedAt()),
-		"updated_at": FormatTime(t.UpdatedAt()),
+		"created_at": FormatTime(t.t0),
+		"updated_at": FormatTime(t.tf),
 		"may_cancel": false,
-		"output":     t.Output(),
+		"output":     t.output,
+	}
+}
+
+var (
+	// ErrNoReceiver is returned from Send when the task isn't listening (yet)
+	ErrNoReceiver = errors.New("task is not receiving messages at this time")
+	// ErrNotRunning is returned from Send when the task isn't Running
+	ErrNotRunning = errors.New("task is not running")
+)
+
+// Send a value into the task
+func (t *Task) Send(v interface{}) error {
+	t.Lock()
+	defer t.Unlock()
+
+	if t.State() != TaskRunning {
+		return ErrNotRunning
+	}
+
+	if t.chin == nil {
+		return ErrNoReceiver
 	}
+
+	t.output = nil
```

Comment:
Why is the output being reset here?  The task and the outside world are completely out of sync. We shouldn't drop output on the floor just because the outside world has sent in some information, right? If it makes sense for the specific task, the task itself may reset its output before consuming the input, but even that feels strange. It means someone asking for the status just after v went in would see nil as the output.